.github/actions/check_version/get_version_from_files.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
```

Comment:
I would separate this script in two different scripts to make it easier to understand and maintain:

1. `get_version_from_os_bin`
2. `get_version_from_firmware_hex`